lib/IPC/Run/Win32Helper.pm

```diff
@@ -444,8 +444,15 @@ sub win32_spawn {
 
    my $process;
    my $cmd_line = join " ", map {
-      ( my $s = $_ ) =~ s/"/"""/g;
-      $s = qq{"$s"} if /[\"\s]|^$/;
+      my $s = $_;
+      if ($s eq "") {
+         $s = '""';
+      } elsif ($s =~ /[\s"]/) {
+         # Double-quotes will be escaped with a backslash, and any backslash
+         # that immediately precedes a double-quote will also be escaped.
+         $s =~ s/(\\*)"/$1$1\\"/g;
```

Comment:
\\* will match 0 or more backslashes. The idea is that we want to put a backslash in front of every double-quote. If the double-quote is not preceded by a backslash this code will simple add a single double-quote. But if there are double-quotes those double-quotes will each be duplicated.

There is no concern of recursion here. It would be wrong to call this regex a second time for the same string as that would add too many escapes anyway. And the nature of /g means that a cursor is always moved forward with each match so there is no chance that this could hang.